core/dnsserver/server_https.go

```diff
@@ -119,6 +119,11 @@ func (s *ServerHTTPS) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// We should expect a packet to be returned that we can send to the client.
 	s.ServeDNS(context.Background(), dw, msg)
 
+	if dw.Msg == nil {
+		http.Error(w, "No response", http.StatusInternalServerError)
```

Comment:
is this the correct response according to the DoH spec?

Reply:
Well, there are no specific requirements for DOH responses with non-successful HTTP status codes. 

The only relevant part of the draft is this:
> DoH clients need to use the same semantic processing of non-successful HTTP status codes as other HTTP clients.

https://tools.ietf.org/html/rfc8484#page-7